### PR TITLE
esdts-improvements

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**

### CODE REVIEW

File: core/constants.go [hunk 1]
```diff
@@ -167,6 +167,10 @@ const (
 	Fungible ESDTType = iota
 	// NonFungible defines the token type for ESDT non fungible tokens
 	NonFungible
+	// NonFungibleV2 defines the token type for ESDT non fungible tokens
+	NonFungibleV2
+	// Meta defines the token type for ESDT meta tokens
+	Meta
```
Comment (sasurobert) [inline core/constants.go:173]: We have SFT as well. Plus bring here the DynamicNFT, DynamicSFT, DynamicMeta as well.

File: core/converters.go [hunk 2]
```diff
@@ -177,3 +177,19 @@ func ConvertToEvenHexBigInt(value *big.Int) string {
...
+// ConvertESDTTypeToUint32 converts the esdt type to uint32
+func ConvertESDTTypeToUint32(esdtType string) (uint32, error) {
+	switch esdtType {
+	case FungibleESDT:
+		return uint32(Fungible), nil
+	case NonFungibleESDT:
+		return uint32(NonFungible), nil
+	case NonFungibleESDTv2:
+		return uint32(NonFungibleV2), nil
+	case MetaESDT:
```
Comment (sasurobert) [inline core/converters.go:190]: add the more types, dynamics as well.

File: core/constants.go (hunk 1 above)
Comment (BeniaminDrasovean) [reply core/constants.go:173]: Added.

File: core/converters.go (hunk 2 above)
Comment (BeniaminDrasovean) [reply core/converters.go:190]: Done.

File: core/constants.go (hunk 1 above)
Comment (sasurobert) [reply core/constants.go:206]: change the order of semiFungible and Meta - maybe add MetaFungible - to have a complete name.

File: core/converters.go (hunk 2 above)
Comment (sasurobert) [reply core/converters.go:190]: MetaFungibleESDT

**Review by sasurobert — APPROVED** [review]

File: core/converters.go [hunk 3]
```diff
@@ -177,3 +177,32 @@ func ConvertToEvenHexBigInt(value *big.Int) string {
 
 	return str
 }
+
+// ConvertESDTTypeToUint32 converts the esdt type to uint32
+func ConvertESDTTypeToUint32(esdtType string) (uint32, error) {
```
Comment (iulianpascalau) [inline core/converters.go:182]: tests?

File: core/constants.go (hunk 1 above)
Comment (iulianpascalau) [reply core/constants.go:206]: +1

Comment (BeniaminDrasovean) [reply core/constants.go:206]: Done.

File: core/converters.go (hunk 3 above)
Comment (BeniaminDrasovean) [reply core/converters.go:182]: Added.

**Review by iulianpascalau — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request merged**